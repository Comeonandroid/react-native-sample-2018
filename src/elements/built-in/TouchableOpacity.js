import React, { PureComponent } from 'react'
import styled from 'styled-components'
import { PropTypes } from 'prop-types'
import withPreventDebounce from '../../utils/HOC/withPreventDebounce'


class TouchableOpacity extends PureComponent {
  render() {
    const { children, ...rest } = this.props
    return (
      <Element {...rest}>{children}</Element>
    )
  }
}

TouchableOpacity.propTypes = {
  onPress: PropTypes.func,
  children: PropTypes.node
}

const Element = styled.TouchableOpacity``

export default withPreventDebounce(TouchableOpacity)