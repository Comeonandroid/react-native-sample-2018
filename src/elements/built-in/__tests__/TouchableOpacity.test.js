import 'react-native'
import React from 'react'
import { itBehavesLike } from '../../../specs/shared_examples/index'

import TouchableOpacity from '../TouchableOpacity'

describe('<TouchableOpacity />', () => {
  itBehavesLike('aThemedComponent', { subject: <TouchableOpacity /> })
  itBehavesLike('aComponentWithChildren', { subject: <TouchableOpacity /> })
})
