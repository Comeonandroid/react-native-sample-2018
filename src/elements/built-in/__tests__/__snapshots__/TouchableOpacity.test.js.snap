// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TouchableOpacity /> dark: renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
/>
`;

exports[`<TouchableOpacity /> light: renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
/>
`;

exports[`<TouchableOpacity /> renders component when passed in 1`] = `
<Styled(TouchableOpacity)>
  <Component />
</Styled(TouchableOpacity)>
`;

exports[`<TouchableOpacity /> renders text when passed in 1`] = `
<Styled(TouchableOpacity)>
  Test
</Styled(TouchableOpacity)>
`;
